src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,700 @@
+        /// <param name="format">A span containing the characters that represent a standard or custom format string that defines the acceptable format for <paramref name="destination"/>.</param>
+        /// <param name="provider">An optional object that supplies culture-specific formatting information for <paramref name="destination"/>.</param>
+        /// <returns></returns>
+        public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan<char> format = default, IFormatProvider? provider = null)
+        {
+            return Number.TryFormatHalf(this, format, NumberFormatInfo.GetInstance(provider), destination, out charsWritten);
+        }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static explicit operator Half(float value)
+        {
+            const int SingleMaxExponent = 0xFF;
+
+            uint floatInt = ToUInt32(value);
+            bool sign = (floatInt & float.SignMask) >> float.SignShift != 0;
+            int exp = (int)(floatInt & float.ExponentMask) >> float.ExponentShift;
+            uint sig = floatInt & float.SignificandMask;
+
+            if (exp == SingleMaxExponent)
+            {
+                if (sig != 0) // NaN
+                {
+                    return CreateHalfNaN(sign, (ulong)sig << 41); // Shift the significand bits to the left end
+                }
+                return sign ? NegativeInfinity : PositiveInfinity;
+            }
+
+            uint sigHalf = sig >> 9 | ((sig & 0x1FFU) != 0 ? 1U : 0U); // RightShiftJam
+
+            if ((exp | (int)sigHalf) == 0)
+            {
+                return new Half(sign, 0, 0);
+            }
+
+            return new Half(RoundPackToHalf(sign, (short)(exp - 0x71), (ushort)(sigHalf | 0x4000)));
+        }
+
+        public static explicit operator Half(double value)
+        {
+            const int DoubleMaxExponent = 0x7FF;
+
+            ulong doubleInt = ToUInt64(value);
+            bool sign = (doubleInt & double.SignMask) >> double.SignShift != 0;
+            int exp = (int)((doubleInt & double.ExponentMask) >> double.ExponentShift);
+            ulong sig = doubleInt & double.SignificandMask;
+
+            if (exp == DoubleMaxExponent)
+            {
+                if (sig != 0) // NaN
+                {
+                    return CreateHalfNaN(sign, sig << 12); // Shift the significand bits to the left end
+                }
+                return sign ? NegativeInfinity : PositiveInfinity;
+            }
+
+            uint sigHalf = (uint)ShiftRightJam(sig, 38);
+            if ((exp | (int)sigHalf) == 0)
+            {
+                return new Half(sign, 0, 0);
+            }
+            return new Half(RoundPackToHalf(sign, (short)(exp - 0x3F1), (ushort)(sigHalf | 0x4000)));
+        }
+
+        // -----------------------Start of from-half conversions-------------------------
+        public static implicit operator float(Half value)
```

Comment:
This should be explicit, as per API review.

Reply:
Can't believe I didn't change that :O